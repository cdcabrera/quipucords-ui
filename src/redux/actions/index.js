import { credentialsActions } from './credentialsActions';
import { factsActions } from './factsActions';
import { reportsActions } from './reportsActions';
import { scansActions } from './scansActions';
import { sourcesActions } from './sourcesActions';
import { statusActions } from './statusActions';
import { userActions } from './userActions';

const reduxActions = {
  credentials: credentialsActions,
  facts: factsActions,
  reports: reportsActions,
  scans: scansActions,
  sources: sourcesActions,
  status: statusActions,
  user: userActions
};

export {
  reduxActions as default,
  reduxActions,
  credentialsActions,
  factsActions,
  reportsActions,
  scansActions,
  sourcesActions,
  statusActions,
  userActions
};
