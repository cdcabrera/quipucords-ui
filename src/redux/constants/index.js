import { aboutModalTypes } from './aboutModalConstants';
import { confirmationModalTypes } from './confirmationModalConstants';
import { credentialsTypes } from './credentialsConstants';
import { factsTypes } from './factsConstants';
import { reportsTypes } from './reportsConstants';
import { scansTypes } from './scansConstants';
import { sourcesTypes } from './sourcesConstants';
import { statusTypes } from './statusConstants';
import { toastNotificationTypes } from './toasNotificationConstants';
import { userTypes } from './userConstants';
import { viewTypes } from './viewConstants';
import { viewPaginationTypes } from './viewPaginationConstants';
import { viewToolbarTypes } from './viewToolbarConstants';

const reduxTypes = {
  aboutModal: aboutModalTypes,
  confirmationModal: confirmationModalTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  status: statusTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes,
  view: viewTypes,
  viewPagination: viewPaginationTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  aboutModalTypes,
  confirmationModalTypes,
  credentialsTypes,
  factsTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  statusTypes,
  toastNotificationTypes,
  userTypes,
  viewTypes,
  viewPaginationTypes,
  viewToolbarTypes
};
