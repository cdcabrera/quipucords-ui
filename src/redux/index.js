import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { withTranslation } from 'react-i18next';
import store from './store';
import reduxActions from './actions';
import reduxReducers from './reducers';
import reduxSelectors from './selectors';
import reduxTypes from './constants';

const connectTranslate = (mapStateToProps, mapDispatchToProps) => component =>
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(withTranslation()(component));

const connectRouterTranslate = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(connectTranslate(mapStateToProps, mapDispatchToProps)(component));

const connectRouter = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(
    connect(
      mapStateToProps,
      mapDispatchToProps
    )(component)
  );

export {
  connect,
  connectRouter,
  connectRouterTranslate,
  connectTranslate,
  reduxActions,
  reduxReducers,
  reduxSelectors,
  reduxTypes,
  store
};
