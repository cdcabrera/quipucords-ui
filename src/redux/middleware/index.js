import { createLogger } from 'redux-logger';
import promiseMiddleware from 'redux-promise-middleware';
import { thunk as thunkMiddleware } from 'redux-thunk';
import { multiActionMiddleware } from './multiActionMiddleware';

/**
 * Redux middleware.
 *
 * @type {Array}
 */
const reduxMiddleware = [thunkMiddleware, multiActionMiddleware, promiseMiddleware];

if (process.env.NODE_ENV !== 'production' && process.env.REACT_APP_DEBUG_MIDDLEWARE === 'true') {
  reduxMiddleware.push(createLogger());
}

export {
  reduxMiddleware as default,
  reduxMiddleware,
  createLogger,
  multiActionMiddleware,
  promiseMiddleware,
  thunkMiddleware
};
