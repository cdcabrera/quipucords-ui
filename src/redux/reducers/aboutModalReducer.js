import { aboutModalTypes } from '../constants';
import { reduxHelpers } from '../common/reduxHelpers';

const initialState = {
  show: false
};

const aboutModalReducer = (state = initialState, action) => {
  switch (action.type) {
    case aboutModalTypes.ABOUT_MODAL_SHOW:
      return reduxHelpers.setStateProp(
        null,
        {
          show: true
        },
        {
          state,
          reset: false
        }
      );

    case aboutModalTypes.ABOUT_MODAL_HIDE:
      return reduxHelpers.setStateProp(
        null,
        {
          show: false
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

aboutModalReducer.initialState = initialState;

export { aboutModalReducer as default, initialState, aboutModalReducer };
