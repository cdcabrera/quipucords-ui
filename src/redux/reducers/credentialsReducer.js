import { credentialsTypes } from '../constants';
import { helpers } from '../../common/helpers';
import { reduxHelpers } from '../common/reduxHelpers';
import apiTypes from '../../constants/apiConstants';

const initialState = {
  view: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    credentials: []
  },
  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    credential: null,
    credentialType: '',
    show: false,
    add: false,
    edit: false,
    delete: false
  }
};

const credentialsReducer = (state = initialState, action) => {
  switch (action.type) {
    case credentialsTypes.CREATE_CREDENTIAL_SHOW:
      return reduxHelpers.setStateProp(
        'update',
        {
          show: true,
          add: true,
          credentialType: action.credentialType
        },
        {
          state,
          initialState
        }
      );

    case credentialsTypes.EDIT_CREDENTIAL_SHOW:
      return reduxHelpers.setStateProp(
        'update',
        {
          show: true,
          edit: true,
          credential: action.credential
        },
        {
          state,
          initialState
        }
      );

    case credentialsTypes.UPDATE_CREDENTIAL_HIDE:
      return reduxHelpers.setStateProp(
        'update',
        {
          show: false
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.REJECTED_ACTION(credentialsTypes.ADD_CREDENTIAL):
      return reduxHelpers.setStateProp(
        'update',
        {
          add: true,
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload).message,
          pending: false
        },
        {
          state,
          reset: false
        }
      );

    case reduxHelpers.REJECTED_ACTION(credentialsTypes.DELETE_CREDENTIAL):
    case reduxHelpers.REJECTED_ACTION(credentialsTypes.DELETE_CREDENTIALS):
      return reduxHelpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload).message,
          delete: true,
          pending: false
        },
        {
          state,
          reset: false
        }
      );

    case reduxHelpers.REJECTED_ACTION(credentialsTypes.UPDATE_CREDENTIAL):
      return reduxHelpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload).message,
          pending: false,
          edit: true
        },
        {
          state,
          reset: false
        }
      );

    case reduxHelpers.REJECTED_ACTION(credentialsTypes.GET_CREDENTIALS):
      return reduxHelpers.setStateProp(
        'view',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload).message
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.PENDING_ACTION(credentialsTypes.ADD_CREDENTIAL):
      return reduxHelpers.setStateProp(
        'update',
        {
          pending: true,
          error: false,
          fulfilled: false
        },
        {
          state,
          reset: false
        }
      );

    case reduxHelpers.PENDING_ACTION(credentialsTypes.DELETE_CREDENTIAL):
    case reduxHelpers.PENDING_ACTION(credentialsTypes.DELETE_CREDENTIALS):
      return reduxHelpers.setStateProp(
        'update',
        {
          pending: true,
          delete: true,
          error: false,
          fulfilled: false
        },
        {
          state,
          reset: false
        }
      );

    case reduxHelpers.PENDING_ACTION(credentialsTypes.UPDATE_CREDENTIAL):
      return reduxHelpers.setStateProp(
        'update',
        {
          pending: true,
          error: false,
          fulfilled: false
        },
        {
          state,
          reset: false
        }
      );

    case reduxHelpers.PENDING_ACTION(credentialsTypes.GET_CREDENTIALS):
      return reduxHelpers.setStateProp(
        'view',
        {
          pending: true,
          credentials: state.view.credentials
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.FULFILLED_ACTION(credentialsTypes.ADD_CREDENTIAL):
      return reduxHelpers.setStateProp(
        'update',
        {
          add: true,
          credential: action.payload.data || {},
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.FULFILLED_ACTION(credentialsTypes.DELETE_CREDENTIAL):
    case reduxHelpers.FULFILLED_ACTION(credentialsTypes.DELETE_CREDENTIALS):
      return reduxHelpers.setStateProp(
        'update',
        {
          delete: true,
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.FULFILLED_ACTION(credentialsTypes.UPDATE_CREDENTIAL):
      return reduxHelpers.setStateProp(
        'update',
        {
          credential: action.payload.data || {},
          edit: true,
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.FULFILLED_ACTION(credentialsTypes.GET_CREDENTIALS):
      return reduxHelpers.setStateProp(
        'view',
        {
          credentials: (action.payload.data && action.payload.data[apiTypes.API_RESPONSE_CREDENTIALS_RESULTS]) || [],
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

credentialsReducer.initialState = initialState;

export { credentialsReducer as default, initialState, credentialsReducer };
