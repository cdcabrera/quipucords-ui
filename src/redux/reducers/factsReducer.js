import { factsTypes } from '../constants';
import { helpers } from '../../common/helpers';
import { reduxHelpers } from '../common/reduxHelpers';

const initialState = {
  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    facts: {}
  }
};

const factsReducer = (state = initialState, action) => {
  switch (action.type) {
    case reduxHelpers.REJECTED_ACTION(factsTypes.ADD_FACTS):
      return reduxHelpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload).message
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.PENDING_ACTION(factsTypes.ADD_FACTS):
      return reduxHelpers.setStateProp(
        'update',
        {
          pending: true
        },
        {
          state,
          initialState
        }
      );

    case reduxHelpers.FULFILLED_ACTION(factsTypes.ADD_FACTS):
      return reduxHelpers.setStateProp(
        'update',
        {
          facts: action.payload.data,
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

factsReducer.initialState = initialState;

export { factsReducer as default, initialState, factsReducer };
