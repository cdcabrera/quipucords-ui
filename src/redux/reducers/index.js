import { combineReducers } from 'redux';
import aboutModalReducer from './aboutModalReducer';
import addSourceWizardReducer from './addSourceWizardReducer';
import confirmationModalReducer from './confirmationModalReducer';
import credentialsReducer from './credentialsReducer';
import factsReducer from './factsReducer';
import queryReducer from './queryReducer';
import reportsReducer from './reportsReducer';
import scansReducer from './scansReducer';
import scansEditReducer from './scansEditReducer';
import sourcesReducer from './sourcesReducer';
import statusReducer from './statusReducer';
import toastNotificationsReducer from './toastNotificationsReducer';
import userReducer from './userReducer';
import viewOptionsReducer from './viewOptionsReducer';

const reducers = {
  aboutModal: aboutModalReducer,
  addSourceWizard: addSourceWizardReducer,
  confirmationModal: confirmationModalReducer,
  credentials: credentialsReducer,
  facts: factsReducer,
  query: queryReducer,
  reports: reportsReducer,
  scans: scansReducer,
  scansEdit: scansEditReducer,
  sources: sourcesReducer,
  status: statusReducer,
  toastNotifications: toastNotificationsReducer,
  user: userReducer,
  viewOptions: viewOptionsReducer
};

const reduxReducers = combineReducers(reducers);

export {
  reduxReducers as default,
  reduxReducers,
  aboutModalReducer,
  addSourceWizardReducer,
  confirmationModalReducer,
  credentialsReducer,
  factsReducer,
  queryReducer,
  reportsReducer,
  scansReducer,
  scansEditReducer,
  sourcesReducer,
  statusReducer,
  toastNotificationsReducer,
  userReducer,
  viewOptionsReducer
};
