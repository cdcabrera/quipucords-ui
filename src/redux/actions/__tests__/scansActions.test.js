import promiseMiddleware from 'redux-promise-middleware';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import moxios from 'moxios';
import { scansReducer, scansActionReducer, scansEditReducer } from '../../reducers';
import { scansActions } from '..';

describe('ScansActions', () => {
  const middleware = [promiseMiddleware()];
  const generateStore = () =>
    createStore(
      combineReducers({
        scans: scansReducer,
        scansAction: scansActionReducer,
        scansEdit: scansEditReducer
      }),
      applyMiddleware(...middleware)
    );

  beforeEach(() => {
    moxios.install();

    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
          test: 'success'
        }
      });
    });
  });

  afterEach(() => {
    moxios.uninstall();
  });

  it('Should return response content for addScan method', done => {
    const store = generateStore();
    const dispatcher = scansActions.addScan();

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansEdit;

      expect(response.add).toEqual(true);
      done();
    });
  });

  it('Should return response content for addStartScan method', done => {
    const store = generateStore();
    const dispatcher = scansActions.addStartScan();

    dispatcher(store.dispatch).then(value => {
      expect(value.action.type).toMatchSnapshot('addStartScan');
      done();
    });
  });

  it('Should return response content for getScans method', done => {
    const store = generateStore();
    const dispatcher = scansActions.getScans();

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scans;

      expect(response.view.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for getScanJobs method', done => {
    const store = generateStore();
    const dispatcher = scansActions.getScanJobs('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.jobs.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for getScanJob method', done => {
    const store = generateStore();
    const dispatcher = scansActions.getScanJob('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.job.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for getConnectionScanResults method', done => {
    const store = generateStore();
    const dispatcher = scansActions.getConnectionScanResults('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.connection.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for getInspectionScanResults method', done => {
    const store = generateStore();
    const dispatcher = scansActions.getInspectionScanResults('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.inspection.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for startScan method', done => {
    const store = generateStore();
    const dispatcher = scansActions.startScan('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.start.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for pauseScan method', done => {
    const store = generateStore();
    const dispatcher = scansActions.pauseScan('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.pause.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for cancelScan method', done => {
    const store = generateStore();
    const dispatcher = scansActions.cancelScan('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.cancel.lorem.fulfilled).toEqual(true);
      done();
    });
  });

  it('Should return response content for restartScan method', done => {
    const store = generateStore();
    const dispatcher = scansActions.restartScan('lorem');

    dispatcher(store.dispatch).then(() => {
      const response = store.getState().scansAction;

      expect(response.restart.lorem.fulfilled).toEqual(true);
      done();
    });
  });
});
