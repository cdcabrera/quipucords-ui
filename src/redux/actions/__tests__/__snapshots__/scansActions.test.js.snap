// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScansActions Should return response content for addStartScan method: addStartScan 1`] = `"ADD_START_SCAN_FULFILLED"`;

exports[`ScansActions Should return response content for getScanJob method: getScanJob 1`] = `"GET_SCAN_JOB_FULFILLED"`;

exports[`ScansActions Should return response content for updatePartialScan method: updatePartialScan 1`] = `"UPDATE_SCAN_FULFILLED"`;

exports[`ScansActions Should return response content for updateScan method: updateScan 1`] = `"UPDATE_SCAN_FULFILLED"`;
